### PR TITLE
feat: Support `wait_ip` property

### PR DESCRIPTION
**Pull request by liangwen12year**
Enhancement:  

Add support for the `wait_ip` property, the system will consider
connection activated only when specific IP stack is configured.
This enables flexibility in scenarios such as
IPv6-only networks, where the overall network configuration can still
succeed when IPv4 configuration fails but IPv6 completes successfully.

The `wait_ip` can be configured with the following possible values:
  * "any": System will consider interface activated when any IP stack is
           configured.
  * "ipv4": System will wait IPv4 been configured.
  * "ipv6": System will wait IPv6 been configured.
  * "ipv4+ipv6": System will wait both IPv4 and IPv6 been configured.

Reason:

Result:

Issue Tracker Tickets (Jira or BZ if any):

Resolves: https://issues.redhat.com/browse/RHEL-63026

### CODE REVIEW

File: examples/eth_with_may_fail.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    network_connections:
+      - name: eth0
+        state: up
+        type: ethernet
+        interface_name: eth0
+        ip:
+          address:
+            - 192.0.2.42/30
+            - 2001:db8::23/64
+          may_fail4: True
```
Comment (richm) [inline examples/eth_with_may_fail.yml:15]:
```suggestion
          may_fail4: true
```

File: examples/eth_with_may_fail.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+          may_fail6: True
```
Comment (richm) [inline examples/eth_with_may_fail.yml:16]:
```suggestion
          may_fail6: true
```

File: tests/tasks/assert_may_fail.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  command: >
+    nmcli -f ipv6.may-fail connection show {{ profile }}
+  register: may_fail6
+  ignore_errors: true
+  changed_when: false
+- name: "Assert that may_fail4 is configured as specified"
+  assert:
+    that:
+      - may_fail4.stdout.split(":")[1] | trim
+        == no
```
Comment (richm) [inline tests/tasks/assert_may_fail.yml:19]:
```suggestion
        == "no"
```

File: tests/tasks/assert_may_fail.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    msg: "may_fail4 is configured as
+      {{ may_fail4.stdout.split(':')[1] | trim }}
+      but specified as {{ may_fail4 }}"
+- name: "Assert that may_fail6 is configured as specified"
+  assert:
+    that:
+      - may_fail6.stdout.split(":")[1] | trim
+        == no
```
Comment (richm) [inline tests/tasks/assert_may_fail.yml:27]:
```suggestion
        == "no"
```

**Comment by liangwen12year** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest]

**Comment by spetrosi** [comment]
[citest_bad]

**Comment by spetrosi** [comment]
[citest]

**Comment by tyll** [comment]
It seems more user friendly to use the Nmstate syntax here with `wait_ip: any|ipv4|ipv6|ipv4+ipv6` - https://nmstate.io/devel/yaml_api.html

**Comment by liangwen12year** [comment]
> It seems more user friendly to use the Nmstate syntax here with `wait_ip: any|ipv4|ipv6|ipv4+ipv6` - https://nmstate.io/devel/yaml_api.html

I changed it to align with nmstate schema.

File: examples/eth_with_may_fail.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- name: Configuring may-fail on ethernet connection
```
Comment (richm) [inline examples/eth_with_may_fail.yml:3]: Is this from a different PR?

**Comment by richm** [comment]
> Added support for the may_fail4 and may_fail6 properties, allowing the network configuration to proceed even if the corresponding IP configuration times out. This enables flexibility in scenarios such as IPv6-only networks, where the overall network configuration can still succeed when IPv4 configuration fails but IPv6 completes successfully.

This should be changed to describe `wait_ip`

**Comment by liangwen12year** [comment]
> > Added support for the may_fail4 and may_fail6 properties, allowing the network configuration to proceed even if the corresponding IP configuration times out. This enables flexibility in scenarios such as IPv6-only networks, where the overall network configuration can still succeed when IPv4 configuration fails but IPv6 completes successfully.
> 
> This should be changed to describe `wait_ip`

I updated it, thanks.

**Comment by liangwen12year** [comment]
[citest]

**Review by richm — APPROVED** [review]

**Comment by richm** [comment]
@liangwen12year ready to merge?

**Review by cathay4t — APPROVED** [review]
Approved, through I don't like we are expanding `network_connections`, especially we are introducing nmstate stuff into this NM specific schema.  Use who request this should use nmstate schema instead of request RFE to `network_connections`.

**Comment by liangwen12year** [comment]
> Approved, through I don't like we are expanding `network_connections`, especially we are introducing nmstate stuff into this NM specific schema. Use who request this should use nmstate schema instead of request RFE to `network_connections`.

I actually requested the issue reporter to use the `network_state` variable, but the reporter has not tried it yet, https://github.com/linux-system-roles/network/issues/735#issuecomment-2394097149.  I also updated the ReadMe to reflect that the future of the network role is promoting using `network_state` variable instead of the `network_connections`, hopefully, we will have fewer feature requests in the `network_connections` variable when the feature is available in the `network_state` .

**Pull request merged**